node-red-contrib-botbuilder/msbot-connect.js

```diff
@@ -7,14 +7,47 @@ const { initConnector } = require("./msbot/actions.js");
 const DEFAULT_TYPING_DELAY = 2000;
 const MINIMUM_TYPING_DELAY = 200;
 
+let REPLY_HANDLER = {};
+let server;
+let allowedCallers = '';
+
 // --------------------------------------------------------------------------
 //  NODE-RED
 // --------------------------------------------------------------------------
+const getAllowedCallers = (node, data, config) => {
+  let result = '';
+  const bType = config.botType;
+  const ac = config.allowedCallers;
+  const acType = config.allowedCallersType;
+
+  if (bType === 'rootBot' || bType === 'skillBot') {
+    if (acType === '' || ac === '') {
+      throw new Error('[BotBuilder]  Param allowedCallers missing');
+    }
 
+    switch (acType) {
+      case 'msg':
+        result = data[ac];
+        break;
+      case 'global':
+        result = node.context().global.get(ac);
+        break;
+      case 'json':
+        result = JSON.parse(ac);
+        break;
+      case 'flow':
+        result = node.context().flow.get(ac);
+        break;
+    }
+  }
+  return result;
+};
+ 
 module.exports = function(RED) {
   const register = function(config) {
     RED.nodes.createNode(this, config);
     var node = this;
+    node.status({});
```

Comment:
this allows us reinitializing the status each time when starting the bot builder without error.
In the code, there is only one status 'error' in case of error (Line 72).  So the empty object is used in case of having a 'none-error' start.